usage/usage_test.go

```diff
@@ -0,0 +1,40 @@
+package usage
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestErrors(t *testing.T) {
+	t.Parallel()
+	u := New()
+	require.NoError(t, u.Uint64("test/one", 1))
+	require.NoError(t, u.Uint64("test/two", 1))
+	require.NoError(t, u.Uint64("test/two", 1))
+	require.NoError(t, u.Strings("test/three", "three"))
+	require.NoError(t, u.Strings("test2/one", "one"))
+
+	require.ErrorContains(t, u.Strings("test/one", "one"),
+		"test/one is not []string as expected but uint64")
+	require.ErrorContains(t, u.Uint64("test2/one", 1),
+		"test2/one is not uint64 as expected but []string")
+
+	require.NoError(t, u.Strings("test3", "some"))
+	require.NoError(t, u.Strings("test3/one", "one"))
+
+	m, err := u.Map()
+	require.ErrorContains(t, err,
+		"key test3 was expected to be a map[string]any but was []string")
```

Comment:
Please, don't consider this message blocking, because I don't have any immediate suggestion/alternative.

But, from my perspective, this UI is a bit confusing, cause I'd expect from `u.Strings("test3/one", "one")` point of view, it to fail if later won't be included in the final report. Otherwise, how I'm supposed to know it? 🤔

Reply:
I don't know how to fix this without remaking the map on each call - which might be better :shrug: Just wasn't particularly nice when we had 3 types (number, string or slice, slice of strings) 

Will see if now it works slightly better.